mrp_production_force_bom/models/mrp_production.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, models
+from odoo.exceptions import ValidationError
+
+
+class MrpProduction(models.Model):
+    _inherit = "mrp.production"
+
+    def _check_unexpected_products(
+        self, move_raw_product_ids, bom_product_ids, error_messages
+    ):
+        unexpected_products = move_raw_product_ids - bom_product_ids
+        if unexpected_products:
+            error_messages.append(
+                _("For order '%s': Some components are not part of the BOM.")
```

Comment:
To make the message slightly simpler. Please do the same for other messages.

```suggestion
                _("%s: Some components are not part of the BOM.")
```